buildlib/azure-pipelines-pr.yml

```diff
@@ -47,6 +47,29 @@ stages:
                  echo "##vso[task.complete result=Failed;]"
               fi
             condition: eq(variables['Build.Reason'], 'PullRequest')
+  - stage: Format
+    jobs:
+      # Check that the code is formatted according to the code style guidelines
+      - job: format
+        displayName: format code
+        container: fedora
+        steps:
+          - checkout: self
+            clean: true
+
+          - bash: |
+              git-clang-format --diff remotes/origin/${SYSTEM_PULLREQUEST_TARGETBRANCH} ${BUILD_SOURCEVERSION} > /tmp/patch
```

Comment:
1. pls create local file and not in /tmp since it may not work will with multiple concurrent jobs
2. also pls print the following info for debugging purposes:
   1. generated patch file
   2. revision range which was checked

e.g 
echo "checking code formatting on revision range x..y"
echo "code formatting diff:"
cat format-diff.patch

Reply:
Done